<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta name="description" content="None">
        <meta name="author" content="serine">
        
        <link rel="shortcut icon" href="img/favicon.ico">
        <title>RNAsik pipeline</title>
        <link href="css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="css/font-awesome.min.css" rel="stylesheet">
        <link href="css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
        <link href="css/style.css" rel="stylesheet">
        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
        <![endif]-->

        <script src="js/jquery-1.10.2.min.js" defer></script>
        <script src="js/bootstrap-3.0.3.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body class="homepage">

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
            <div class="container">

                <!-- Collapsed navigation -->
                <div class="navbar-header">
                    <!-- Expander button -->
                    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                        <span class="sr-only">Toggle navigation</span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href=".">RNAsik pipeline</a>
                </div>

                <!-- Expanded navigation -->
                <div class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="active">
                                <a href=".">About</a>
                            </li>
                            <li >
                                <a href="docs/">Docs</a>
                            </li>
                            <li >
                                <a href="tutorials/">Tutes</a>
                            </li>
                            <li >
                                <a href="help/">Help</a>
                            </li>
                            <li >
                                <a href="contrib/">Contributing</a>
                            </li>
                            <li >
                                <a href="roadmap/">Roadmap</a>
                            </li>
                            <li >
                                <a href="checklist/">Checklist</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav navbar-right">
                        <li>
                            <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="disabled">
                                <a rel="next" >
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li >
                                <a rel="prev" href="docs/">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li>
                                <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="main active"><a href="#rnasik-for-rnaseq">RNAsik for RNAseq</a></li>
            <li><a href="#about">About</a></li>
            <li><a href="#how-to-cite">How to cite</a></li>
            <li><a href="#methods">Methods</a></li>
            <li><a href="#mbp-team-photo">MBP team photo</a></li>
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<p><img alt="mbp-banner" src="images/mbp_banner.png" /></p>
<h1 id="rnasik-for-rnaseq">RNAsik for RNAseq</h1>
<blockquote>
<p><strong>N.B</strong> This workflow assumes model organism has a reference genome. If the reference genome isn't applicable, different workflow might be required.</p>
</blockquote>
<h2 id="about">About</h2>
<p><a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe">RNAsik</a> pipeline was build in house for processing <a href="https://rnaseq.uoregon.edu/">RNA-seq(uencing)</a> data.
It is written in <a href="http://pcingola.github.io/BigDataScript/download.html">BigDataScript (bds)</a>, which is <a href="https://en.wikipedia.org/wiki/Domain-specific_language">domain specific language (DSL)</a>, that makes writing pipelines easy as well as making them robust. To get a bit more technical, bds runs on <a href="https://en.wikipedia.org/wiki/Java_virtual_machine">java virtual machine (JVM)</a> and therefore requires <a href="https://www.java.com/en/">Java</a>.</p>
<p>In simple terms any pipeline is a wrapper of several tools that makes it easier and arguably faster to get to the end goal. The three core parts to any <a href="https://rnaseq.uoregon.edu/">RNA-seq analysis</a> are: </p>
<ul>
<li>mapping to the reference genome</li>
<li>counting reads mapped into features e.g genes</li>
<li>doing differential expression (DE) statistics</li>
</ul>
<p>The pipeline does the first two parts and <a href="http://degust.erc.monash.edu">Degust</a> does the third part. Degust itself, in simple terms, a wrapper around <a href="http://bioconductor.org/packages/release/bioc/html/limma.html">limma</a> and <a href="http://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> R packages. In theory and practice one can take output from <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe">RNAsik</a> pipeline, which is a table of counts where every gene is a row and every column is a sample and use those with any other R packages that do DE analysis.</p>
<p>In actual terms both <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe">RNAsik</a> and <a href="http://degust.erc.monash.edu">Degust</a> provide complete experience, not only you'll get your list of DE genes and QC metrics, but will be able to get full inside into your experimental design and the outcome of that. <code>RNAsik</code> does read alignment and read counting and cleaning and improvements of your table of counts, which makes <a href="http://degust.erc.monash.edu">Degust</a> analysis one upload away. <code>RNAsik</code> wraps <a href="docs/#prerequisites">these tools</a> making your RNAseq analysis more streamline. It also has "sanity checks" inbuilt, checking command line options, checking if options are valid files/directories and it will talk to you so don't sweat :) but do read the error messages. <a href="http://degust.erc.monash.edu">Degust</a> is exceptionally good for exploratory data visualisation and analysis. Both tools can also server as a nice proxy for learning bioinformatics as they provide command line and R code for doing the analysis. Last but not least thanks to <a href="http://multiqc.info/">MultiQC</a> <code>RNAsik</code> provides an aggregate of different metrics in one place - multiqc report. This is a good place to start understanding your data.</p>
<p>The central bits of information are:</p>
<ul>
<li>Are there differences in library sizes?</li>
<li>Is there any issues with mapping rates?</li>
<li>Is there any issues with reads assignment rates?</li>
</ul>
<p>However there is so many other questions you can ask including:</p>
<ul>
<li>What is duplication rate?</li>
<li>What is multi-mapping rate?</li>
<li>What is intragenic and interagenic rates?</li>
</ul>
<p>As mentioned above <a href="http://multiqc.info">multiqc</a> report is a great first step in the attempt to answer those questions. A lot of the time everything looks fairly good and consistent allowing downstream analysis. Sometimes user can tweak certain individual parameters which can improve results, other times it comes down to experimental design and/or library preparation and sequencing issues. Either way one need to make this "first iteration" in order to see room for improvement. </p>
<h2 id="how-to-cite">How to cite</h2>
<p><a href="https://doi.org/10.5281/zenodo.1403976"><img alt="DOI" src="https://zenodo.org/badge/DOI/10.5281/zenodo.1403976.svg" /></a></p>
<hr />
<p>Tsyganov, Kirill, Andrew James Perry, Stuart Kenneth Archer, and David Powell. 2018. “RNAsik: A Pipeline for Complete and Reproducible RNA-Seq Analysis That Runs Anywhere with Speed and Ease.” Journal of Open Source Software 3: 583.</p>
<hr />
<p><em>It is hard to give full acknowlegment to all contributors. The nature of the open source projects such that contributors can come and go, however they leave behind valuable contributions and need to get full credit for that. Please look at <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe">RNAsik GitHub</a> repository to get a full sense of who is contributing. In particular one can look at <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/graphs/contributors">number of commits</a>, <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/issues">issues triaging and handling</a> and <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/pulls">pull requests (PRs)</a>. Please also remember that every contribution matters, nothing is too small!</em></p>
<h2 id="methods">Methods</h2>
<p>Raw fastq files have been analysed with RNAsik pipeline <span class="citation">(Tsyganov et al. 2018)</span> to produce raw genes count matrix and various quality control metrics. For this analysis RNAsik pipeline <span class="citation">(Tsyganov et al. 2018)</span> ran with STAR aligner option <span class="citation">(Dobin et al. 2013)</span> and reads were quantified with featureCounts <span class="citation">(Liao, Smyth, and Shi 2014)</span>. The reference GTF and FASTA files were downloaded from <a href="https://ensembl.org">Ensembl database</a>. Raw counts were then analysed with Degust <span class="citation">(Powell 2015)</span> web tool to do differential expression analysis to produce list of differentially expressed genes and several quality plots including classical multidimensional scaling (MDS) and MA plots. In this analysis limma voom <span class="citation">(Law et al. 2014)</span> was used for differential expression analysis. Degust <span class="citation">(Powell 2015)</span> largely follows limma voom workflow with typical conts per million (CPM) library size normalisation and trimmed mean of M values (TMM) normalisation <span class="citation">(Robinson and Oshlack 2010)</span> for RNA composition normalisation.</p>

<h2 id="references" class="unnumbered">References</h2>

<div id="refs" class="references">
<div id="ref-Dobin2013-yw">
<p>Dobin, Alexander, Carrie A Davis, Felix Schlesinger, Jorg Drenkow, Chris Zaleski, Sonali Jha, Philippe Batut, Mark Chaisson, and Thomas R Gingeras. 2013. “STAR: Ultrafast Universal RNA-seq Aligner.” <em>Bioinformatics</em> 29 (1): 15–21. <a href="http://dx.doi.org/10.1093/bioinformatics/bts635" class="uri">http://dx.doi.org/10.1093/bioinformatics/bts635</a>.</p>
</div>
<div id="ref-Law2014-ev">
<p>Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. “Voom: Precision Weights Unlock Linear Model Analysis Tools for RNA-seq Read Counts.” <em>Genome Biol.</em> 15 (2): R29. <a href="http://dx.doi.org/10.1186/gb-2014-15-2-r29" class="uri">http://dx.doi.org/10.1186/gb-2014-15-2-r29</a>.</p>
</div>
<div id="ref-Liao2014-qo">
<p>Liao, Yang, Gordon K Smyth, and Wei Shi. 2014. “FeatureCounts: An Efficient General Purpose Program for Assigning Sequence Reads to Genomic Features.” <em>Bioinformatics</em> 30 (7): 923–30. <a href="http://dx.doi.org/10.1093/bioinformatics/btt656" class="uri">http://dx.doi.org/10.1093/bioinformatics/btt656</a>.</p>
</div>
<div id="ref-Powell2015">
<p>Powell, David. 2015. “Degust: Powerfull and User Friendly Front-End Data Analsysis, Visualisation and Exploratory Tool for Rna-Sequencing.” github. <a href="http://degust.erc.monash.edu" class="uri">http://degust.erc.monash.edu</a>.</p>
</div>
<div id="ref-Robinson2010-yu">
<p>Robinson, Mark D, and Alicia Oshlack. 2010. “A Scaling Normalization Method for Differential Expression Analysis of RNA-seq Data.” <em>Genome Biol.</em> 11 (3): R25. <a href="http://dx.doi.org/10.1186/gb-2010-11-3-r25" class="uri">http://dx.doi.org/10.1186/gb-2010-11-3-r25</a>.</p>
</div>
<div id="ref-Tsyganov2018-si">
<p>Tsyganov, Kirill, Andrew James Perry, Stuart Kenneth Archer, and David Powell. 2018. “RNAsik: A Pipeline for Complete and Reproducible RNA-seq Analysis That Runs Anywhere with Speed and Ease.” <em>Journal of Open Source Software</em> 3: 583. <a href="https://www.theoj.org/joss-papers/joss.00583/10.21105.joss.00583.pdf" class="uri">https://www.theoj.org/joss-papers/joss.00583/10.21105.joss.00583.pdf</a>.</p>
</div>
</div>

<h2 id="mbp-team-photo">MBP team photo</h2>
<p><img alt="team_photo_2017" src="images/team_photo_2017.jpg" /></p>
<p><a href="https://twitter.com/intent/tweet?screen_name=kizza_a" class="twitter-mention-button" data-size="large" data-show-count="false">Tweet to @kizza_a</a><script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script> </p>

<p class="twitter-btn">
<a class="twitter-share-button"
  href="https://twitter.com/intent/tweet?text=Hey%20I%27m%20using%20this%20fully%20sick%20RNAseq%20pipeline%20It%27s%20sik%20easy%20http%3A%2F%2Fgithub%2Ecom%2Fmonashbioinformaticsplatform%2FRNAsik%2Dpipe%20by%20%40kizza%5Fa%20from%20%40MonashBioinfo" data-size="large">
Share</a>
</p></div>
        </div>

        <footer class="col-md-12">
            <hr>
                <p>Created by <a href="https://twitter.com/kizza_a">Kirill Tsyganov</a> with help from the <a href="http://bioinformatics.erc.monash.edu">Monash Bioinformatics Platform</a> team. <br /><a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.</p>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = ".",
                shortcuts = {"help": 191, "search": 83, "previous": 80, "next": 78};
        </script>
        <script src="js/base.js" defer></script>
        <script src="search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Search</h4>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form role="form">
                    <div class="form-group">
                        <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="Keyboard Shortcuts Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Keyboard Shortcuts</h4>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>

<!--
MkDocs version : 1.0.4
Build Date UTC : 2019-10-15 11:52:42
-->
