---
title: 'RNAsik: Pipeline for complete and reproducible RNA-seq analysis that runs anywhere with speed and ease'
tags:
- RNA-seq
- gene expression
- pipeline
- bioinformatics
authors:
- name: Kirill Tsyganov
  orcid: 0000-0002-9195-2626
  affiliation: 1
- name: Andrew James Perry
  orcid: 0000-0001-9256-6068
  affiliation: 1
- name: Stuart Archer
  orcid: 
  affiliation: 1
- name: David Powell
  orcid: 0000-0002-6639-6250
  affiliation: 1
affiliations:
- name: 'Monash Bioinformatics Platform, Monash University'
  index: 1
date: 6 February 2018
bibliography: paper.bib
---

# Summary

RNA-sequencing (RNA-seq) is one of the applications of high throughput sequencing. Illumina based sequencing can produce millions of short reads, typically around 100 bases long. RNAsik pipeline makes processing of RNA-seq data streamline and gives reproducible results. You can run RNAsik standalone or on HPC cluster. Single RNAsik run gives comprehensive overview of your experiment and sets you up for DGE calling, which is most commonly used application for RNA-seq.
In an alignment based approach, three main steps are; alignment of short reads contained in FASTQ files to a reference genome, counting the number of reads mapped to the genomic features (genes) and calling DGE. The table of counts generated by RNAsik can be further analysed with any other DGE tools/packages that take this common input format - one such package particularly suited to the task is Degust [@Powell2015], a powerful front-end and user friendly tool for DGE data analysis, visualisation and exploration. Additionally RNAsik includes a number of different quality control (QC) metrics; short read quality metrics reported using FastQC [@Bioinformatics2011-ve], intra- and inter-genic rates estimation using QualiMap [@Okonechnikov2016-jk], library size and GC bias estimation using Picard Tools [@picard-tools2018]. These QC metrics are automatically summarised in a single, dynamic, html report generated by MulitQC [@Ewels2016-ik]. Other nice features to note; RNAsik improves table of counts with addition of extra meta information about each gene e.g biotype and human readable gene names, it provides ready to use coverage plots for every sample using bedtools2 [@Quinlan2010-am] and UCSC tools [@Raney2014-ti], and RNAsik logs every step in the pipeline including number of samples and associated FASTQ files, tooling versions and sequencing strand information. 
RNAsik is written in BigDataScript (BDS) [@Cingolani2015-nm], which is a domain specific language (DSL). BDS generates an additional html report about your RNAsik run, unlike RNAsik internal logging, this report holds system information such as run-time information and exit status for every tool. BDS has many other great features such as inbuilt checkpointing for retries on failure and ability to talk to the HPC cluster queue directly. 
RNAsik includes STAR aligner [@Dobin2013-yw], featureCounts [@Liao2014-qo], samtools [@Li2009-aa] to name some of the tools. However, one can extend RNAsik with other new tools and features. Recently, RNAsik had been extend to include two other aligners, Hisat2 [@Kim2015-ju] and BWA-MEM [@Li2013-bz]. This broadens the scope of RNAsik to bacterial RNA-seq analysis and improves diversity.
RNAsik is an open-source project under Apache License 2.0, any contributions are welcomed. In the near future, there are plans to extend RNAsik in several directions including alignment free reads quantification as well as including RNA-seq variant calling option [@Sun2016-ud]. RNAsik simplifies and speeds up your RNA-seq analysis enabling and automates many of the QC steps that are important but often overlooked.
