<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="serine">
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Roadmap - RNAsik pipeline</title>
        <link href="../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
        <link href="../css/style.css" rel="stylesheet">
        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
        <![endif]-->

        <script src="../js/jquery-1.10.2.min.js" defer></script>
        <script src="../js/bootstrap-3.0.3.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
            <div class="container">

                <!-- Collapsed navigation -->
                <div class="navbar-header">
                    <!-- Expander button -->
                    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                        <span class="sr-only">Toggle navigation</span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href="..">RNAsik pipeline</a>
                </div>

                <!-- Expanded navigation -->
                <div class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li >
                                <a href="..">About</a>
                            </li>
                            <li >
                                <a href="../docs/">Docs</a>
                            </li>
                            <li >
                                <a href="../tutorials/">Tutes</a>
                            </li>
                            <li >
                                <a href="../help/">Help</a>
                            </li>
                            <li >
                                <a href="../contrib/">Contributing</a>
                            </li>
                            <li class="active">
                                <a href="./">Roadmap</a>
                            </li>
                            <li >
                                <a href="../checklist/">Checklist</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav navbar-right">
                        <li>
                            <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li >
                                <a rel="next" href="../contrib/">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li >
                                <a rel="prev" href="../checklist/">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li>
                                <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="main active"><a href="#roadmap">Roadmap</a></li>
            <li><a href="#going-forward">Going forward</a></li>
            <li><a href="#ideas-for-future-releases">Ideas for future releases</a></li>
            <li><a href="#changelog">Changelog</a></li>
            <li><a href="#content">Content</a></li>
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<p><img alt="mbp-banner" src="../images/mbp_banner.png" /></p>
<h1 id="roadmap">Roadmap</h1>
<h2 id="going-forward">Going forward</h2>
<h3 id="1xx">1.x.x</h3>
<ul>
<li>need to have better way to check for index directory, want to know if starIdx includes or doesn't indexing with annotation.</li>
<li>recheck <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/pull/10/commits/9e64da57de6da066e94bf6fcc66e23c36adb3671">Stuart's PR</a>, polish off refFiles detection</li>
<li>recheck <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/pull/10">this whole PR</a></li>
<li>noticed that qualimap could have high RAM consumption, need to fix cpu and mem parameters passing through sik.config file. Reckon to set mem at 4 or 6 Gb</li>
<li>add support for another aligner - <a href="https://github.com/lh3/minimap2">minimap2</a></li>
</ul>
<h3 id="154-end-of-the-year-2018">1.5.4 (end of the year) 2018</h3>
<ul>
<li>include handling of url based samples sheets, i.e <code>-samplesSheet</code> flag should handle local based or remote files, just like <code>-fastaRef</code> option</li>
<li>better tools version logging, don't like when <code>RNAsik</code> checks <code>bwa</code> version when <code>STAR</code> aligner is used</li>
<li>include logging of split lanes and R1 and R2. Want to be able to see from the log whether two reads were classified as split lanes or paired end. This is
to do with recent bug that got fixed in <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/commit/b9240274fa7c964e953a767c254f31ba0d044547">b924027</a></li>
<li>add alignment free support for RNAseq analysis salmon/kalisto</li>
</ul>
<h3 id="160-februarymarch-2019">1.6.0 (February/March) 2019</h3>
<ul>
<li>
<p>Plans to add variants calling to <code>RNAsik</code>. It'll be opt in flag,  <code>-varsCall</code>. Suggestions are welcomed about different name for a flag.
I already have a prototype in bds, just need to plug it in.</p>
<ul>
<li>Not sure which caller to use <code>GATK</code> or <code>freebayes</code> will need to do more reading on that.</li>
<li>Also need to document common pitfalls for using RNAseq for variant calling e.g can only can variances in coding regions that are expressed.</li>
<li>Not a good idea to use pulled samples as you won't be able to get allele frequency</li>
<li>will also need to find out where to get known SNP's (snpDB? for known germ line mutations) and a list of blacklisted regions</li>
</ul>
</li>
<li>
<p><a href="https://github.com/CRG-CNAG/CalliNGS-NF">an example of someone else variant calling pipeline</a></p>
</li>
</ul>
<h2 id="ideas-for-future-releases">Ideas for future releases</h2>
<ul>
<li>Need better support for exonic/intronic rates estimation. Is <code>read_distribution.py</code> from RSeQC good idea? Right now qualiMap is ok flag to opt in.</li>
<li>is there need for circular RNA support?</li>
<li>add demultiplexign tools into pipeline; <code>-demult</code>, <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/issues/32">more info here</a></li>
</ul>
<h2 id="changelog">Changelog</h2>
<h3 id="152">1.5.2</h3>
<p>features:</p>
<ul>
<li><code>-fqDir</code> can now take file with links to fastq files, links can have local or remote paths</li>
<li>coverage plots are now stranded, i.e for every bam file you'll get forward, reverse and all (non) stranded coverages</li>
<li>trimming has now support for both single and paired-end data</li>
<li>started to include logging of individual tools, started with sikQC.bds file. Basically redirecting individual tools stderr/stdout into logs files.</li>
</ul>
<p>maintenance:</p>
<ul>
<li>include continue integration testing - TraviCI and started writing more test that cover bds source as well as additional scripts source</li>
<li>bug fixes and improvements of python scripts.</li>
<li>added testing for <code>get_geneids.py</code> script, this isn't python unit test rather simple "does the tool run" type test.</li>
<li>introduced short args notation for python scripts as an option to long args naming</li>
<li>pulled <a href="https://github.com/pansapiens"><a class="citation" href="#ref-pansapiens" id="cite-pansapiens">pansapiens</a></a> PR, now we have <code>RNASIK_BDS_CONFIG</code> variable for passing bds.config</li>
<li>fixed samtools sort bug that led to out-of-memory kill</li>
<li>all (most?) picard tools now have mem and cpu parameters</li>
<li>fixed bwa mem dependencies issue, for <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/blob/b5acbd03c0b79323420caead79927c6c4e00f92e/src/sikSTARaligner.bds#L62">more info here</a></li>
<li>fixed qualimap issue with DISPLAY variable, explicitly unsetting it now</li>
<li>fixed pairIds overide, only set pairIds to default value _R1,_R2 if no -pairIds given on cmd</li>
<li>removed paired-end checking from inside bwa, all checks happens outside of alinger scope now</li>
<li>improved multiqc report visual, by tweaking config file, the report shouldn't look as cluttered</li>
</ul>
<p>conda:</p>
<ul>
<li>started packaging bds and RNAsik with conda, packages are available at <a href="https://anaconda.org/serine">my anaconda channel</a></li>
<li>updated docs with how to conda build, install and upload</li>
<li>wrote quick_install.bash script that does install and used in traviCI</li>
</ul>
<h3 id="151">1.5.1</h3>
<ul>
<li>several bug fixes including <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/issues/18">#18</a>, hisat2 related bugs, and samtools sort memory bug</li>
<li>in relation to bug <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/issues/18">#18</a> added extra <a href="https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/blob/b3f91ca72e2c34a97f3d6b757a2fc3b9027df001/src/sikUtils.bds#L153">sanity check</a></li>
<li>general code improvement and maintenance</li>
<li>updated docs and included new installation method</li>
<li>improved python scripts, made them user executable and updated docs with how to use them</li>
<li>updated UCSC binaries due to libpng12 issue</li>
</ul>
<h3 id="150">1.5.0</h3>
<ul>
<li>fixed bugs and improved python scripts, also migrated them to python3</li>
<li>fixed issues: #16, #15, #20, #21</li>
<li>fixed bug in handling remote reference files</li>
<li>added <code>mk_igv_links.py</code> script as general utility script</li>
<li>added FASTQ trimmer - <code>skewer</code></li>
<li>internal code improvements, made code more readable and clean</li>
<li>Re-wrote bam files processing, i.e sorting and marking duplicates. Changed sorting from <code>picard SortSam</code> to <code>samtools sort</code>. Kept <code>picard MarkDuplicates</code> for conventional marking duplicates, but add <a href="https://gbcs.embl.de/portal/tiki-index.php?page=Je">Je-suite</a> for UMI based de-duplication.</li>
<li>Made improvements in handling of featureCounts output for multiqc purposes.</li>
<li>completely removed threads and memory parameters, now each task's gets it's own cpu and mem setting all through sik.config. This makes it more cluster friendly as well</li>
<li>
<p>Added more metrics gathering; samtools qc metrics:</p>
<ul>
<li><code>flagstat</code></li>
<li><code>idxstats</code></li>
<li><code>stats</code></li>
</ul>
</li>
<li>
<p>Improved many different task's dependencies flow.</p>
</li>
<li>
<p>changed few command lines options; new options:</p>
<ul>
<li><code>-all</code></li>
<li><code>-counts</code></li>
<li><code>-mdups</code></li>
<li><code>-qc</code></li>
<li><code>-umi</code></li>
<li><code>-trim</code></li>
</ul>
</li>
<li>
<p>removed command line options:</p>
<ul>
<li><code>-metrics</code></li>
<li><code>-fastqc</code></li>
<li><code>-prePro</code></li>
<li><code>-threads</code></li>
<li><code>-memory</code></li>
</ul>
</li>
<li>
<p>all flags changes have some backward compatibility, i.e will set the closet new option.</p>
</li>
</ul>
<h3 id="149">1.4.9</h3>
<ul>
<li>Fixed BWA indexing problem. Problem around STAR and Hisat2 aligners pass index as directory whereas bwa as a file, had to fight to make all different options i.e <code>-refFiles</code> and <code>-genomeIdx</code> to work</li>
<li>Included proper support for SAF file format handled through -gtfFile flag i.e user can pass GTF, GFF or SAF through that flag and will still get counts</li>
<li>Imporved handling of urls for refrence files <code>-fastaRef</code>, <code>-gtfFile</code> now works with all bds supported url types. Also fixed tarball url handling through <code>-fqDir</code></li>
<li>Improved code readability in several places and included <code>canFail</code> flag for making degust reads counts files.</li>
<li>Fixed a bug in <code>exonicRates</code> function was passing "wrong" gtf file path</li>
<li>Updated docs, added more explanation on how RNAsik ticks. Included a roadmap to allow better time and features management</li>
<li>fixed <code>-pairIds</code> bug, courtesy @stu2 (PR #6) and samples sheet file making</li>
<li>Improved python script, including several small bug fixing</li>
</ul>
<h3 id="148">1.4.8</h3>
<ul>
<li>added new feature: coverage plots generation</li>
<li>added new feature: ability to pass previously generated references directory, saves spaces and time</li>
<li>fixed strand guessing script, should be better at guessing now</li>
<li>moved to <a href="http://www.mkdocs.org/">mkdocs</a> for documents compilation and deployment to gh_pages</li>
<li>added new python script to make degust file, this simplifies and strengthens the code</li>
<li>improved readability of the code</li>
<li>improved and fixed bugs in handling fastq files and assignment of fastq to sample names</li>
</ul>
<h3 id="147">1.4.7</h3>
<ul>
<li>fixed STAR memory allocation issue, now user can run with fewer cpus without a worry for STAR spawning multiple tasks causing out of memory issue.</li>
<li>made BASH wrapper (not ideal) for <code>RNAsik</code> this is capture <code>bds</code> logs including report.html which is rather valuable piece of information about the run</li>
<li>introduced another new aligner - <code>bwa mem</code> to do bacterial RNAseq.</li>
<li>introduced multiqc config file in <code>configs</code> directory now</li>
<li>completely removed <code>-fqRegex</code> and added sanity check for paired end data. If R2 is found and -paired isn't set or vice verse then error message sent</li>
<li>several general bug fixes</li>
<li>simplified help menu</li>
</ul>
<h3 id="146">1.4.6</h3>
<ul>
<li>generalised aligner's call, this will make it easier to add new aligners into RNAsik</li>
<li>included support for <code>hista2</code>aligner</li>
<li>added samplesSheep logging</li>
<li>fixed few minor bugs and improved code quality</li>
<li>added <code>canFail</code> option to several non crucial tasks, allowing pipeline to continue if some task failed of fastqc and qualimap to allow them to fail as those are non essential tasks.</li>
</ul>
<h3 id="13">1.3</h3>
<h2 id="content">Content</h2>
<ul>
<li><a href="#general">General</a></li>
<li><a href="#star">STAR</a></li>
<li><a href="#featureCounts">featureCounts</a></li>
</ul>
<h4 id="general">General</h4>
<ul>
<li>Improved documentations and help menu</li>
<li>Introduced new <code>-extraOptions</code> command. Now user can pass in any additional commands that particular tools can have. This will make pipeline rather flexible.</li>
<li>
<p>Included more checks along the way, such as:</p>
<ul>
<li>checking if directories and files given on the cli are valid and accessible</li>
<li>check if FASTQ files where found in the given location</li>
<li>if no <code>-align</code> option was specified check that <code>bamFiles</code> directory exists and has BAM files</li>
<li>if no <code>-prePro</code> options specified checks for <code>preqcBamFiles</code> directory and not present through an error. Can now use <code>-proBams</code> for processed bams to specify RNAseQC ready bams</li>
</ul>
</li>
<li>
<p>Made <code>RNAsik-pipe</code> more sequential, i.e step one run STAR, then run featureCounts then run processing steps. Previously featureCounts and picard tools were running in paralell and I/O was suffering. It is faster to have featureCounts end its run before doing picard processing.</p>
</li>
<li>
<p>Removed <code>-makeIndicies</code> and <code>-fqDirs</code> options and simplified input</p>
<ul>
<li>Combined <code>-fqDir</code> and <code>fqDirs</code> into single flag <code>-fqDir</code> under which you can specify either directory with FASTQ's or a project directory with directories with FASTQ's. In another way, now pipeline looks two level deep for FASTQ files in a given directory under <code>-fqDir</code></li>
<li><code>-makeIndex</code> now only makes STAR index. In future it might accept aligner name for which to make an index</li>
<li>other indicies for picard and RNAseQC are now made on the fly when those tools are called</li>
</ul>
</li>
<li>Don't have to specify <code>-makeIndex</code> if running an alignment. If <code>-align</code> given and no <code>-genomeIndex</code> is specified, then pipeline will ask for <code>-fastaRef</code> and make index before aligning</li>
<li>Improved scoping. RNAsik-pipe is a mine file that runs everything, all other files define functions and don't look outside of the file. Everything is passed in from within main file - RANsik-pipe.</li>
<li>Included an option to specify gzipped files, but not in the <code>-extraOptions</code> if additoinal file is need to be given through <code>extraOptions</code> and file is gzipped it won't be handled. The gzipped option is specific for <code>fastaRef</code> and <code>gtfFile</code> only. Those can be gzipped and pipieline will handle it.</li>
<li>also now copy <code>fastaRef</code> and <code>gtfFile</code> into <code>refFiles</code> directory. This is to avoid permission issues. Because original destination might not have all the required files, such as <code>.dict</code> and <code>.fai</code> files and if not permitted pipeline won't be able to execute certain steps.</li>
<li>Introduced <code>-sampleNames</code> options now user can "fix" BAM file names and all the downstream file and data naming to something more minegful.</li>
</ul>
<h4 id="star">STAR</h4>
<ul>
<li>Changed bool <code>-star</code> option for <code>-align</code> that now accepts a string. At the moment the only string it can accept is <code>star</code> (lower case). This is again step towards making pipeline more flexible for users demands. In future I'm planing to introduce other aligners to my pipeline so that user perhaps can choose an aligner.</li>
<li>Auto detect is FASTQ are paired-end or not. Wrote a function that loops through <code>-fqDir</code> directory and counting all the R1 and R2 reads. Based on the counts it ditermines whether data is paired end or not.</li>
</ul>
<h4 id="featurecounts">featureCounts</h4>
<ul>
<li>included a couple of python script that uses <code>gffutils</code> package to make a datatbase of GTF entries and then parse that database file to get public gene names, Ensmebl Ids and gene's biotypes. It is easy to include more information about the gene with this python module.</li>
<li>Included more post processing on count files, now they are ready for degust upload</li>
</ul>
<h3 id="12">1.2</h3>
<ul>
<li>
<p>Changed the way <code>RNAsik-pipe</code> gets read counts files. Now it creates only two files one for reverse
stranded data and another for non stranded data. featureNo and featureReverse directories with
corresponding read counts files have been removed and replaced with featureNo.txt and
featureRevere.txt that already will have all of your reads in columns as expected by
<a href="http://www.vicbioinformatics.com/degust/">Degust</a>. Minor cosmetic changes are made to both
featureNo.txt and featureReverse.txt to convert then into featureNoCounts.txt and
featureReverseCounts.txt respectively. Use counts file for your
<a href="http://www.vicbioinformatics.com/degust/">Degust</a> session.</p>
</li>
<li>
<p>Added <code>threads</code> option, default at [1], so that user can specify number of threads they wish to
  use.</p>
</li>
<li>
<p>Added <code>makeIndex</code> option, which allows user to either make all indices with <code>makeIndices</code>, that is
  <code>STAR genomeGenerate</code>, <code>picard CreateSequnceDictionary</code> and <code>samtools faidx</code>, but given that STAR index
takes a long time, can be some hours if running on minimal number of threads and user already has
BAM files and just wishes to use other modules of the <code>RNAsik-pipe</code>, user can now just make other
required indices.</p>
</li>
</ul>
<p><a href="https://twitter.com/intent/tweet?screen_name=kizza_a" class="twitter-mention-button" data-size="large" data-show-count="false">Tweet to @kizza_a</a><script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script> </p>

<p class="twitter-btn">
<a class="twitter-share-button"
  href="https://twitter.com/intent/tweet?text=Hey%20I%27m%20using%20this%20fully%20sick%20RNAseq%20pipeline%20It%27s%20sik%20easy%20http%3A%2F%2Fgithub%2Ecom%2Fmonashbioinformaticsplatform%2FRNAsik%2Dpipe%20by%20%40kizza%5Fa%20from%20%40MonashBioinfo" data-size="large">
Share</a>
</p>

<div class="references">
<table>
<tbody>
<tr id="ref-pansapiens">
<td>pansapiens</td>
<td>Missing citation</td>
</tr>
</tbody>
</table>
</div></div>
        </div>

        <footer class="col-md-12">
            <hr>
                <p>Created by <a href="https://twitter.com/kizza_a">Kirill Tsyganov</a> with help from the <a href="http://bioinformatics.erc.monash.edu">Monash Bioinformatics Platform</a> team. <br /><a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.</p>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "..",
                shortcuts = {"previous": 80, "help": 191, "next": 78, "search": 83};
        </script>
        <script src="../js/base.js" defer></script>
        <script src="../search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Search</h4>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form role="form">
                    <div class="form-group">
                        <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="Keyboard Shortcuts Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Keyboard Shortcuts</h4>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
